olive/data_config/container/base_container.py

```diff
@@ -0,0 +1,83 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from typing import ClassVar
+
+from pydantic import BaseModel
+
+from olive.data_config.component.dataloader import default_calibration_dataloader
+from olive.data_config.config import DataConfig, DefaultDataComponentCombos
+from olive.data_config.constants import DataContainerType, DefaultDataContainer
+from olive.data_config.registry import Registry
+
+
+@Registry.register(DataContainerType.DATA_CONTAINER, name=DefaultDataContainer.DATA_CONTAINER.value)
+class BaseContainer(BaseModel):
```

Comment:
>BaseContainer [](http://example.com/codeflow?start=6&length=13)

could we rename the container to dataset here? Just like huggingface dataset, it has its own dataset and torch dataset as well.

Reply:
It the base class for data container which includes four components. `Dataset` seems a little confused with huggingface Dataset concept.

Reply:
As `dataset` focus on the dataset itself without tokenizer, post process and etc.